docs/architecture/adr-006-hermes-v0.2-usecases.md

```diff
@@ -0,0 +1,249 @@
+# ADR 006: Hermes v0.2.0 Use-Cases
+
+## Changelog
+* 16.02.2021: Proposed.
+
+## Context
+
+One major problem with planning for the evolution of Hermes is that presently
+there is insufficient clarity regarding its requirements.
+It is not known who are the typical Hermes users (is it human operators or
+automated pipelines?), and what are their primary use-cases.
+
+This ADR proposes a few use-cases that seem interesting from the point
+of view of a general target base of users, and which will
+hopefully be a subset of the requirements of (any) future users.
+
+Three elements that provide further context for this discussion are:
+
+1. Hermes is still at an early stage of implementation, so these use-cases are
+   not set in stone.
+
+2. Some concrete use-cases are starting to emerge ([#628][628]), which Hermes
+   v0.1.0 either does not cover altogether, or covers poorly (e.g., because of
+   inconsistent UX), thus informing this proposal.
+
+3. Hermes is one of _three_ relayer binaries that are being developed roughly in
+parallel. The other two are being developed in Go and Typescript, 
+respectively (see the [references](#references) section).
+In this context, it is plausible that Hermes will focus on performance,
+robustness, and richness of features on a longer term.
+
+## Decision
+
+This is a summary of the use-cases (commands) discussed in the rest of this ADR.
+Note that the commands below omit the binary name `hermes` , to keep the command
+length to a minimum.
+
+To create and update a client:
+- `create client ibc-0 ibc-1 create client ibc-0 ibc-1 [--clock-drift <millis>] [--trusting-period <days>] [--trust-threshold <numerator/denominator>]`
+- `update client ibc-0 <client-id>`
+
+To create a connection:
+- `create connection ibc-0 ibc-1 [--delay <delay>]`
+- `create connection ibc-0 --src-client <client-id> --dst-client <client-id> [--delay <delay>]`
+
+To create a channel:
+- `create channel ibc-0 ibc-1 --src-port <port-id> --dst-port <port-id> [--order <order>] [--version <version>] [--delay <delay>]`
+- `create channel ibc-0 --src-port <port-id> --dst-port <port-id> --src-connection <connection-id> [--order <order>] [--version <version>]`
+
+To start packet relaying:
+- `start ibc-0 ibc-1 --src-port <port-id> --dst-port <port-id> [--order <order>] [--version <version>] [--delay <delay>]`
+- `start ibc-0 --src-port <port-id> --dst-port <port-id> --src-connection <connection-id> [--order <order>]
+  [--version <version>]`
+- `start ibc-0 --src-channel <channel-id>`
+
+For finishing pre-initialized, but unfinished object handshakes, for connection and channel:
+- `establish connection ibc-0 --src-connection <connection-id>`
+- `establish channel ibc-0 --src-channel <channel-id>`
```

Comment:
I think we need the port to disambiguate the channel
```suggestion
- `establish channel ibc-0 --src-port <port-id> --src-channel <channel-id>`
```

Reply:
Good point. Added it. d6c95c72